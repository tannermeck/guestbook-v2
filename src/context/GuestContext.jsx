import { useContext, useState, createContext } from 'react';

const GuestContext = createContext();

const GuestProvider = ({children}) => {
    const [guest, setGuest] = useState('')

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
const useGuest = () => {
    const context = useContext(GuestContext);
    if (context === undefined){
        throw new Error('useGuest hook must be called inside of a GuessContext Provider');
    }
    return context;
}
export { GuestProvider, useGuest}