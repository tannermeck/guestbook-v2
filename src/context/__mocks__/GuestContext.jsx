import { createContext, useState } from "react";

const GuestContext = createContext();

const GuestProvider = ({mockUser, children}) => {
    const [guest, setGuest] = useState(
    mockUser ? 'tanner' : '' )

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
export { GuestProvider, GuestContext }