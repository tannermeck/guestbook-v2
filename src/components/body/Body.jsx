import { useState } from 'react';
import { useGuest } from '../../context/GuestContext';
import Entry from '../entry/Entry';
import './body.css'

const Body = () => {
    const {guest, setGuest} = useGuest();
    const [name, setName] = useState('')
    const [entry, setEntry] = useState('')
    const [entries, setEntries] = useState([])

    const handleGuestEntry = (e) => {
        e.preventDefault();
        setGuest(name)
        setEntries([...entries, {name, entry}])
        setEntry('')
    }

    return (
        <div className='inputField'>
            <div>
                {!guest && 
                <h3>Please Sign the Guestbook!</h3> 
                }
                {guest && 
                    <h3>{`Thanks for signing ${guest}`}</h3>
                }
            </div>
            <form onSubmit={handleGuestEntry}>
                {!guest &&
                <div className='text'>
                    <label>Guest Name</label>
                    <input type='text' placeholder='Guest Name...' value={name} onChange={(e) => setName(e.target.value)} required/>
                </div>
                }
                <div className='text'>
                    <label>Guest Entry</label>
                    <textarea placeholder='Your Entry' value={entry} onChange={(e) => setEntry(e.target.value)} required/>
                </div>
                <div className='buttonAndSignout'>
                    <button>Sign</button>
                    {guest && 
                        <button className='signout'
                            onClick={()=> {
                                setGuest('')
                                setName('')
                                setEntry('')
                            }}
                        >
                            {`Not ${guest} ?`}</button>
                    }
                </div>
            </form>
            <Entry entries={entries} />
        </div>
    )
}
export default Body;