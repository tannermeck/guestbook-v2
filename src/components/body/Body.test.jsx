import {screen, render, findByText } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { GuestProvider } from '../../context/GuestContext';
import Body from './Body';

it('should render the body component', async() => {
    const { container } = render(
                            <GuestProvider>
                                <Body />
                            </GuestProvider>)

    const nameInput = await screen.findByPlaceholderText('Guest Name...');
    const entryInput = await screen.findByPlaceholderText('Your Entry');
    const button = await screen.findByRole('button', {name: 'sign'})

    userEvent.type(nameInput, 'Tanner');
    userEvent.type(entryInput, 'Thanks');
    userEvent.click(button);
    
    const nameEntry = await screen.findByText('Thanks for signing Tanner')
    const guestEntry = await screen.findByText('Thanks');

    expect(nameEntry).toBeInTheDocument();
    expect(guestEntry).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});