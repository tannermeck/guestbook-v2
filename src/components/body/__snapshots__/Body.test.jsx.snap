// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the body component 1`] = `
<div>
  <div
    class="inputField"
  >
    <div>
      <h3>
        Please Sign the Guestbook!
      </h3>
      
    </div>
    <form>
      <div
        class="text"
      >
        <label>
          Guest Entry
        </label>
        <textarea
          placeholder="Your Entry"
          required=""
        />
      </div>
      <div
        class="buttonAndSignout"
      >
        <button
          aria-label="sign"
        >
          Sign
        </button>
        
      </div>
    </form>
    <section
      class="entryContainer"
    />
  </div>
</div>
`;
